pandas/core/index.py

```diff
@@ -508,24 +515,26 @@ def best_len(values):
             else:
                 return 0
 
+        data = self._formatter_data
```

Comment:
@jreback This is because ``Index._formatter_func`` converts categorical to int like below. 


```
i = pd.CategoricalIndex(pd.period_range('2011-01', freq='M', periods=3))
i._formatter_func(i[:2])
# u'[492, 493]'
```

I think we should fix it either by:

- Change passed data (current impl)
- Change ``Index.format_data`` to handle categorical
- Change ``com.pprint_thing`` 

Which is preferable?